handler/ezBIDS_core/ezBIDS_core.py

```diff
@@ -1129,15 +1143,23 @@ def datatype_suffix_identification(dataset_list_unique_series):
 
                 # Assume not BIDS-compliant acquisition, unless user specifies otherwise
                 else:
-                    unique_dic["error"] = " ".join("Acquisition cannot be resolved. Please \
-                        determine whether or not this acquisition should be \
-                        converted to BIDS".split())
-                    unique_dic["message"] = " ".join("Acquisition is unknown because there \
-                        is not enough adequate information, primarily in the \
-                        SeriesDescription. Please modify if acquisition is desired \
-                        for BIDS conversion, otherwise the acquisition will not be \
-                        converted".split())
-                    unique_dic["type"] = "exclude"
+                    # okay this hack needs work but it's a start
+                    if unique_dic.get("sidecar", {}).get("Modality","") == "PT":
+                        #unique_dic["message"] = "Acquisition is believed to be PET"
+                        #unique_dic["type"] = "pet"
+                        #unique_dic["datatype"] = "pet"
+                        #unique_dic["suffix"] = "pet"
+                        pass
+                    if unique_dic.get("sidecar", {}).get("Modality","") != "PT":
+                        unique_dic["error"] = " ".join("Acquisition cannot be resolved. Please \
+                            determine whether or not this acquisition should be \
+                            converted to BIDS".split())
+                        unique_dic["message"] = " ".join("Acquisition is unknown because there \
+                            is not enough adequate information, primarily in the \
+                            SeriesDescription. Please modify if acquisition is desired \
+                            for BIDS conversion, otherwise the acquisition will not be \
+                            converted".split())
+                        unique_dic["type"] = "exclude"
```

Comment:
So, this was a bit of a bust in making sure that PET files didn't end up as excluded. @dlevitas what values is this dictionary looking for so that one doesn't end up with the following:

<img width="1211" alt="Screen Shot 2023-05-01 at 1 48 33 PM" src="https://user-images.githubusercontent.com/28850131/235701393-10324f69-459a-4f3b-a23a-f16279d6ebfa.png">

I can keep plodding along, but it's probably faster to ask.

Reply:
@bendhouseart were you able to progress smoothly up to this point (i.e., the `Series Mapping` and `Dataset Review` pages contained the correct PET information such as the `datatype` and `suffix`)? If so, then the issue is probably the [convert.js file](https://github.com/brainlife/ezbids/blob/master/handler/convert.js), which converts the contents of the `ezBIDS_core.json` to an appropriate BIDS mapping. 

I assume you're working on localhost so I can ask for your ezBIDS url

Reply:
Alternatively, could you email me your `ezBIDS_core.json`? On the first page, there's a `Debug` button at the button that allows you to download the `ezBIDS_core.json`

Reply:
Thanks for the file contents. A few things:
1). It appears that each acquisition's `SeriesDescription` is being listed as a subject ID (`sub` entity label). I assume this is because the data has been anonymized, due to `NA` values in fields such as `PatientName` and `PatientID`. ezBIDS can have trouble determining subject (and session) mapping when data is anonymized, we typically ask users to store data in an explicitly marked subject (and session) folder to make this mapping easier (e.g., `sub-01`). Otherwise, you'll have to manually adjust this on the `Subjects/Sessions` web page. 

2). The `type` fields are listed as "/", these should contain the datatype/suffix format (e.g., `pet/pet`). I realize ezBIDS doesn't offer `pet/pet` as a manually choice, so I'll need to add that to the dropdown options, but if your PET converter is able to discern this information, it can be added there.

Lastly, for the `convert.js` file, we'll need to add a `function handlePET()` or something like that.

Reply:
@bendhouseart I've added the PET capabilities to the ezBIDS front end, see my most recent commit (5546d28), which adds some of the PET features in my earlier response. I don't know PET well, so we may need to add some features to `funnction handlePET()`, but this can be our start point.

Reply:
1) Hmm, well I am using phantoms so the patient name and ID are accurate. Would it be useful to add an `--ezbids` flag to pet2bids such that `PatientID`, `PatientName`, and `SeriesDescription` are injected back into the sidecar json? Additionally, we could tailor the output filename a bit if that's helpful too. Will do some testing with properly pre-labeled folders to see what shakes out too, e.g. (`sub-*/ses-*`). I'm of the same opinion as you that a user should make some effort to organize their data in order to have the converter work.

2) I think managed to get that working, but either way I'll integrate the changes you pushed the via 5547d28 into my branch and go from there.

Thanks,

Anthony

Reply:
It just seemed strange to me that each file had a different subject ID, unless that was part of your test. Regardless, we should only worry about including new fields in the json sidecars if the BIDS specification requires it, such as `TaskName` for functional BOLD data, which is also handled by convert.js.

Reply:
I think it's on me, but to be fair I've been doing my best imitation of a user. I've got a bunch of subfolders that I've been dragging and dropping into ezBIDS, however the parent is labeled in a more human readable form. See `source/GeneralElectricAdvance-NIMH`

Reply:
Hmm, no such lucking add your changes to branch. Now getting:

```
/tmp/ezbids-workdir/6451727fbdb6f2becc427ed2/bids/GeneralElectricAdvance-NIMH
|-- README
|-- dataset_description.json
|-- finalized.json
|-- participants.json
`-- participants.tsv

0 directories, 5 files
```

Think this may be the issue:

```+ ./convert.js /tmp/ezbids-workdir/64517553bdb6f2becc427f24
unknown datatype:pet/pet
unknown datatype:pet/pet
unknown datatype:pet/pet
unknown datatype:pet/pet
+ echo 'output bids directory structure'
+ tree /tmp/ezbids-workdir/64517553bdb6f2becc427f24/bids/2d_unif_lt_ramp
+ echo 'running bids validator'
+ bids-validator /tmp/ezbids-workdir/64517553bdb6f2becc427f24/bids/2d_unif_lt_ramp
(node:2134) Warning: Closing directory handle on garbage collection
(Use `node --trace-warnings ...` to show where the warning was created)
+ true
``` 

But that's for future me to figure out.

Reply:
Strange, I did a quick test with `sub-SiemensHRRTNRU` and I was able to have it pass validation. Two questions:
1). When you reach the `Series Mapping` page is the `Datatype/Suffix` field already set to `pet/pet` or do you have to select it from the dropdown menu?
2). It looks like you made some recent additions to the convert.js and .ts files, could you comment those out just for now?